input/input_socket.go

```diff
@@ -94,16 +113,36 @@ func (si *SocketInput) handleServerConnection() {
 	}
 }
 
+func (si *SocketInput) sendPerfStats() {
+	start := time.Now()
+	for {
+		select {
+		case <-si.StopChan:
+			return
+		default:
+			if time.Since(start) > perfStatsSendInterval {
```

Comment:
I don't understand why you just time.Sleep(perfStatsSendInterval) here. Can you add a comment here, please!

Reply:
Because `time.Sleep()` is not interruptible via a stop signal on `si.StopChan`, for example. This would mean that if we want to stop the pipeline, we would possibly have to wait for a while `perfStatsSendInterval` before the loop gets another chance to check that case.
With the current construct we briefly wake up once a second to check whether we are asked to stop or whether it's time to submit stats, which is neglegible in overhead but massively improves shutdown time.

Reply:
Added a comment in the code as well.